pywt/src/common.h

```diff
@@ -64,29 +66,29 @@ typedef enum {
  * Length of DWT coeffs for specified input data length, filter length and
  * signal extension mode.
  */
-index_t dwt_buffer_length(index_t input_len, index_t filter_len, MODE mode);
+size_t dwt_buffer_length(size_t input_len, size_t filter_len, MODE mode);
 
-/* 
+/*
  * Length of reconstructed signal for specified input coeffs length and filter
  * length. It is used for direct reconstruction from coefficients (normal
  * convolution of upsampled coeffs with filter).
  */
-index_t reconstruction_buffer_length(index_t coeffs_len, index_t filter_len);
+size_t reconstruction_buffer_length(size_t coeffs_len, size_t filter_len);
 
-/* 
+/*
  * Length of IDWT reconstructed signal for specified input coeffs length, filter
  * length and extension mode.
  */
-index_t idwt_buffer_length(index_t coeffs_len, index_t filter_len, MODE mode);
+size_t idwt_buffer_length(size_t coeffs_len, size_t filter_len, MODE mode);
 
 /* Length of SWT coefficients for specified input signal length (== input_len) */
-index_t swt_buffer_length(index_t input_len);
+size_t swt_buffer_length(size_t input_len);
 
 /* Maximum useful level of DWT decomposition. */
-int dwt_max_level(index_t input_len, index_t filter_len);
+unsigned char dwt_max_level(size_t input_len, size_t filter_len);
 
 /* Maximum useful level of SWT decomposition. */
-int swt_max_level(index_t input_len);
+unsigned char swt_max_level(size_t input_len);
```

Comment:
My reasoning was that the `input_len` required for a particular level is `2**level`. `2**64 - 1` is the upper bound for a `size_t` (on 64-bit architectures), so more is not required. Otherwise no objections to the change.